pkg/services/state.go

```diff
@@ -236,3 +237,13 @@ func (eb *ErrorBuffer) SetCap(cap int) {
 	}
 	eb.cap = cap
 }
+
+// EnsureClosed closes the io.Closer, returning nil if it was already
+// closed or not started yet
+func EnsureClosed(c io.Closer) error {
+	err := c.Close()
+	if errors.Is(err, ErrAlreadyStopped) || errors.Is(err, ErrCannotStopUnstarted) {
```

Comment:
Was this being used in production?
I thought it was only a hacky workaround for old problems which no longer exist, and so dropping it would be ideal, but limiting to tests would be an improvement.

Reply:
As far as I can tell it's currently used by the [txm](https://github.com/smartcontractkit/chainlink/blob/develop/common/txmgr/txmgr.go#L393), nowhere else. I wasn't aware of the problem we faced in the past.

Reply:
The main problem in the past had to do with test helpers, so I was surprised to see this outside of tests.

Reply:
I've added a fix on the main repo